scanpipe/pipelines/docker.py

```diff
@@ -92,6 +99,32 @@ def collect_and_create_system_packages(self):
             for image in self.images:
                 docker.scan_image_for_system_packages(self.project, image)
 
+    def run_scancode(self):
+        """
+        Scans extracted codebase/ content.
+        """
+        self.scan_output = self.project.get_output_file_path("scancode", "json")
+
+        with self.save_errors(scancode.ScancodeError):
+            scancode.run_scancode(
+                location=str(self.project.codebase_path),
+                output_file=str(self.scan_output),
+                options=self.scancode_options,
+                raise_on_error=True,
+            )
+
+        if not self.scan_output.exists():
+            raise FileNotFoundError("ScanCode output not available.")
+
+    def build_inventory_from_scan(self):
+        """
+        Processes the JSON scan results to determine resources and packages.
+        """
+        project = self.project
+        scanned_codebase = scancode.get_virtual_codebase(project, str(self.scan_output))
+        scancode.create_discovered_packages(project, scanned_codebase)
+        scancode.create_codebase_resources(project, scanned_codebase)
+
```

Comment:
This is a copy paste and it does not respect the DRY principle.
We need a better way to add those common steps to Pipeline subclasses.

Reply:
I removed the code I copied into the `Docker` pipeline and instead, I've added `scan_package.ScanPackage` as a superclass to the `Docker` pipeline.

https://github.com/nexB/scancode.io/blob/d867c52fc941414662191458944f03536f155416/scanpipe/pipelines/docker.py#L29

Is this an acceptable solution or should we create a more generic scan pipeline that can be used as the base for other scan-related pipelines?